File: 2022/16/main.go
package main

import (
	"fmt"
	"io/ioutil"
	"strings"
)

type valve struct {
	name          string
	pressureValue int
	linkedValves  []*valve
}

func main() {
	f, _ := ioutil.ReadFile("inputtest")
	content := strings.Split(string(f), "\n")

	// valves := make(map[string]*valve)

	for _, line := range content {
		line = strings.ReplaceAll(line, "Valve ", "")
		line = strings.ReplaceAll(line, "has flow rate=", " ")
		line = strings.ReplaceAll(line, "; tunnels lead to valves", " ")
		line = strings.ReplaceAll(line, ",", "")
		vals := strings.Split(line, " ")
		fmt.Println(vals)
	}
}

File: 2022/16/main.py
class valve:
    def __init__(self, name, pressure_val):
        self.name = name
        self.pressure_val = pressure_val
        self.valve = []

    def add_valve(self, valve):
        self.valve.append(valve)


if __name__ == "__main__":
    valves = []
    with open("inputtest") as f:
        content = f.readlines()
        for c in content:
            c = c.strip()
            